pkgs/by-name/ma/marytts/package.nix

```diff
@@ -0,0 +1,60 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  # Gradle 8 complains about implicit task dependencies when using `installDist`.
+  # See marytts/marytts#1112
+  gradle_7,
+  makeWrapper,
+  jdk,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "marytts";
+  version = "5.3-unstable-2024-10-09";
+
+  src = fetchFromGitHub {
+    owner = "marytts";
+    repo = "marytts";
+    rev = "1c2aaa0751b7cef8ae83216dd78b4c61232b3840";
+    hash = "sha256-jGpsD6IwJ67nDLnulBn8DycXCyowssSnDCkQXBIfOH8=";
+  };
+
+  nativeBuildInputs = [
+    gradle_7
+    makeWrapper
+  ];
+
+  mitmCache = gradle_7.fetchDeps {
+    inherit (finalAttrs) pname;
+    data = ./deps.json;
+  };
+
+  __darwinAllowLocalNetworking = true;
+
+  gradleBuildTask = "installDist";
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    mv build/install/source/{lib,user-dictionaries} $out
+
+    makeWrapper ${lib.getExe jdk} $out/bin/marytts-server \
+      --add-flags "-cp \"/var/lib/marytts/lib/*:$out/lib/*\" -Dmary.base=/var/lib/marytts" \
+      --append-flags "marytts.server.Mary"
+
+    # We skip the GUI installer since frankly it is a PITA to get to work with a hardened systemd service,
+    # and the imperative installation paradigm is not ideal either way while using Nix.
+
+    runHook postInstall
+  '';
+
```

Comment:
You have the nixos test but you could add it here `passthru.tests.nixos = nixosTests.marytts;`